kernel/core/utils/stringtools.cpp

```diff
@@ -999,8 +1034,7 @@ static bool isDateTimePattern(const std::string& sStr, size_t pos)
 /// \return bool
 ///
 /////////////////////////////////////////////////
-bool isConvertible(const std::string& sStr, ConvertibleType type)
-{
+bool isConvertible(const std::string& sStr, ConvertibleType type){

```

Comment:
Consider the following adaption of this function:

```suggestion
bool isConvertible(const std::string& sStr, ConvertibleType type, NumberFormatsVoter* voter = nullptr){
```

Here, `NumberFormatsVoter` is a structure/class to be designed replacing your two global variables. It may also contain the `vote()` method from further up and will only be used if non-zero.

```C++
struct NumberFormatsVoter
{ // Preferring ANSI brace format ;-)
    // necessary attributes here

    void vote(int numType)
    {
        // ...
    }

    int getFormat() const
    {
        // ...
    }
}
```